### PR TITLE
Document the extensibility of the elasticsearch format for products and product models

### PR DESCRIPTION
**Pull request by mmetayer**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: internal_doc/enrichment/extend_search_engine_format.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+services:
+    Acme\Bundle\MyCustomBundle\Elasticsearch\Product\AddExternalId:
+        arguments:
+            - ...
+        tags:
+            - { name: akeneo.pim.enrichment.product.query.indexing_additional_properties }
+```
+
+## Append a property the product model projection
```
Comment (SamirBoulil) [inline internal_doc/enrichment/extend_search_engine_format.md]:
```suggestion
## Append a property to the product model projection
```

File: internal_doc/enrichment/extend_search_engine_format.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        //...
+    ],
+  ]
+```
+The indexed values must be json encodable, meaning they can be either
+scalars, arrays of scalars or objects (implementing the
+`\JsonSerializable` interface is a bonus). For more information see
+https://www.php.net/manual/en/function.json-encode.php
+
+- You can inject several
```
Comment (SamirBoulil) [inline internal_doc/enrichment/extend_search_engine_format.md]:
```suggestion
- You can inject several properties with
```

Comment (mmetayer) [reply internal_doc/enrichment/extend_search_engine_format.md]: nope, that's not what I meant, but it's not clear enough, I'll rephrase

File: internal_doc/enrichment/extend_search_engine_format.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+As of v4.0, the indexation of a product in Elasticsearch is now
+performed via a single entrypoint: the
+`Akeneo\Pim\Enrichment\Bundle\Elasticsearch\Model\ElasticsearchProductProjection`.
+Its implementation is quite naive: a constructor which sets all of its
+properties, and a `toArray()` method called by the ProductIndexer in
+order to build the document that will be sent to Elasticsearch. The same
+logic is applied to the indexation of product models, with
+`Akeneo\Pim\Enrichment\Bundle\Elasticsearch\Model\ElasticsearchProductModelProjection`.
+
+Thess classe are `final`, meaning we do not mean them to be extended nor
```
Comment (SamirBoulil) [inline internal_doc/enrichment/extend_search_engine_format.md]: I didn't know Thess was a word, (I didn't find its definition :/)

Comment (mmetayer) [reply internal_doc/enrichment/extend_search_engine_format.md]: I meant "These classes" :sweat_smile:

**Pull request merged**